docs/source/blueprints/BP000.rst

```diff
@@ -0,0 +1,336 @@
+   better overview of it.
+
+ * Motivation: This should be the motivation of your proposed solution, not the
+   motivation of the blueprint itself. It describes the problem. Here, you
+   should answer "why" your solution is needed.
+
+ * Specification: In this section, you should describe how you are going to
+   solve the problem. You can create subsections here to organize your ideas
+   better. Please keep in mind that it is useful to mention the details, with
+   code snippets, examples, and/or references. This will save you time, making
+   sure that everyone is in agreement with the proposed solution.
+
+ * Backwards Compatibility: How is your proposal going to affect older versions
+   of Avocado? Should we deprecate some modules, classes, or methods? Are we
+   going to keep backwards compatibility or not?
+
+ * Security Implications: Do you have any concerns about security with your
+   proposed solution and what are they?
+
+ * How to Teach This: What is the best way to inform our devs and users about
+   your new feature/solution?  feature/solution?
+
+ * Related Issues: Here, you should mention Github links for both: a) current
+   open issues that are blocking while waiting for your BP and b) all open
+   issues that will render this BP as “implemented” when closed.
+
+   1. Issues to address this BP
+
+      Would be nice, if possible, to open issues on GH that covers all aspects
+      of your Blueprint.
+
+   2. Issues this BP will solve
+
+      What are the issues already existent on Avocado project that your
+      proposal will solve?
+
+ * References: Any external reference for helping understand the problem and
+   your solution.
+
+Backwards Compatibility
+***********************
+
+So far, we are on our 3rth blueprint (BP003 was the last one). This BP000
+should have been released before those blueprints. So probably those three
+blueprints are not 100% compliance with this meta blueprint, and that is fine.
+We were learning on the fly. We don’t need to change any of those blueprints
+after BP000 gets approved.
+
+Security Implications
+*********************
+
+No security implications found so far.
+
+How to Teach This
+*****************
+
+Getting used to writing blueprints is not an easy task. And probably we are
+going to find unplanned issues with this process on the way.  The general
+rule of thumb is to use common sense. To make this more public, we could
+consider the following:
+
+ * If approved, BP000 should be on top of our blueprints lists for reference.
+
+ * We could also have a template inside the `blueprints` directory to help
+   people when submitting their own blueprints.
+
+ * Also, we could include pointers and instructions in our development guide
+   for this BP.
+
+ * Another good practice would be to make comments in Avocado’s source code
+   with some pointers to specific blueprints.
+
+Related Issues
+**************
+
+None yet.
+
+References
+**********
+
+Todo
```

Comment:
Is this meant as a document called TODO which is the reference or as "references are still to be done"?

Reply:
Here is to add references for your BP. But in this case, we don't have, so I will remove this word. THanks